cache/in-memory/src/lib.rs

```diff
@@ -691,6 +691,16 @@ impl InMemoryCache {
     }
 
     fn cache_role(&self, guild_id: GuildId, role: Role) -> Arc<Role> {
+        // Insert the role into the guild_roles map
+        if let Some(mut g) = self.0.guild_roles.get_mut(&guild_id) {
+            g.insert(role.id);
+        } else {
+            let mut set = HashSet::new();
+            set.insert(role.id);
+            self.0.guild_roles.insert(guild_id, set);
+        }
```

Comment:
I think this is more readable
```suggestion
        self.0.guild_roles
            .entry(&guild_id)
            .or_default()
            .insert(role.id);
```

Reply:
That is indeed, I did not know about the entry API, ill commit this.